lib/new_relic/telemetry/broadway.ex

```diff
@@ -0,0 +1,190 @@
+defmodule NewRelic.Telemetry.Broadway do
+  @moduledoc false
+  use GenServer
+
+  alias NewRelic.Transaction
+  alias NewRelic.DistributedTrace
+
+  require Logger
+
+  @processor_start [:broadway, :processor, :start]
+  @processor_stop [:broadway, :processor, :stop]
+
+  @message_stop [:broadway, :processor, :message, :stop]
+  @message_fail [:broadway, :processor, :message, :failure]
+
+  @consumer_start [:broadway, :consumer, :start]
+  @consumer_stop [:broadway, :consumer, :stop]
+
+  def start_link() do
+    enabled = true
+    GenServer.start_link(__MODULE__, [enabled: enabled], name: __MODULE__)
+  end
+
+  def init(enabled: false), do: :ignore
+
+  def init(enabled: true) do
+    config = %{
+      handler_id: :new_relic_broadway
+    }
+
+    :telemetry.attach_many(
+      config.handler_id,
+      [
+        @processor_start,
+        @processor_stop,
+        @message_stop,
+        @message_fail,
+        @consumer_start,
+        @consumer_stop
+      ],
+      &__MODULE__.handle_event/4,
+      config
+    )
+
+    Process.flag(:trap_exit, true)
+    {:ok, config}
+  end
+
+  def terminate(_reason, %{handler_id: handler_id}) do
+    :telemetry.detach(handler_id)
+  end
+
+  def handle_event(@processor_start, _measurements, metadata, _config) do
+    Transaction.start_transaction("Broadway", processor_name(metadata))
+    NewRelic.add_attributes(processor_start_attributes(metadata))
+
+    DistributedTrace.generate_new_context()
+    |> DistributedTrace.track_transaction(transport_type: "HTTP")
+  end
+
+  def handle_event(@processor_stop, _measurements, metadata, _config) do
+    NewRelic.add_attributes(processor_stop_attributes(metadata))
+    Transaction.stop_transaction()
+  end
+
+  def handle_event(@message_stop, measurements, metadata, _config) do
+    track_message_segment(measurements, metadata)
+  end
+
+  def handle_event(@message_fail, measurements, metadata, _config) do
```

Comment:
If there was already an exception raised & caught, it's possible this error state is already handled. We'll need to verify this with tests of that condition

Reply:
Here goes a problem: Broadway captures all the errors and logs then, but it is not clear to me if they end on `error_logger`. Maybe @msaraiva may help, or we can open an issue on Broadway repo?

Reply:
Yeah, if broadway rescues them, they probably don't make it too logger at all. We'll want to exercise this in a simple test though

Reply:
Hi folks!

All errors, unless explicitly wrapped in a `try/catch` block, should be captured by Broadway and logged using `Logger.error/2`. In case there's any scenario you have which is not covered and you need any additional feature, please feel free to open an issue at https://github.com/dashbitco/broadway for discussion.

Cheers.